Gems/Atom/Feature/Common/Code/Source/RayTracing/RayTracingPass.cpp

```diff
@@ -364,92 +500,94 @@ namespace AZ
                 }
 
                 m_rayTracingShaderTable->Build(descriptor);
-            }
-        }
-
-        void RayTracingPass::BuildCommandListInternal(const RHI::FrameGraphExecuteContext& context)
-        {
-            RPI::Scene* scene = m_pipeline->GetScene();
-            RayTracingFeatureProcessor* rayTracingFeatureProcessor = scene->GetFeatureProcessor<RayTracingFeatureProcessor>();
-            AZ_Assert(rayTracingFeatureProcessor, "RayTracingPass requires the RayTracingFeatureProcessor");
 
-            if (!rayTracingFeatureProcessor ||
-                !rayTracingFeatureProcessor->GetTlas()->GetTlasBuffer() ||
-                !rayTracingFeatureProcessor->HasGeometry() ||
-                !m_rayTracingShaderTable)
-            {
-                return;
+                // register the shader-table with the dispatch item
+                m_dispatchRaysItem.SetRayTracingPipelineState(m_rayTracingPipelineState.get());
+                m_dispatchRaysItem.SetRayTracingShaderTable(m_rayTracingShaderTable.get());
             }
 
-            RHI::DeviceDispatchRaysItem dispatchRaysItem;
-
-            // calculate thread counts if this is a full screen raytracing pass
-            if (m_passData->m_makeFullscreenPass)
+            // Collect and register the Srgs (RayTracingGlobal, RayTracingScene, ViewSrg, SceneSrg and RayTracingMaterialSrg)
+            // The more consistent way would be to call BindSrg() of the RenderPass, and then call
+            // SetSrgsForDispatchRays() in BuildCommandListInternal, but that function doesn't exist.
+            // [GFX TODO][ATOM-15610] Add RenderPass::SetSrgsForRayTracingDispatch
+            if (m_shaderResourceGroup != nullptr)
             {
-                RPI::PassAttachment* outputAttachment = nullptr;
-
-                if (GetOutputCount() > 0)
-                {
-                    outputAttachment = GetOutputBinding(0).GetAttachment().get();
-                }
-                else if (GetInputOutputCount() > 0)
+                auto constantIndex = m_shaderResourceGroup->FindShaderInputConstantIndex(Name("m_maxRayLength"));
+                if (constantIndex.IsValid())
                 {
-                    outputAttachment = GetInputOutputBinding(0).GetAttachment().get();
+                    m_shaderResourceGroup->SetConstant(constantIndex, m_maxRayLength);
                 }
```

Comment:
There is a helper class called ShaderInputNameIndex that can help simplify code like this and save the name and index value so they don't need to be re-initialized each frame.

Reply:
done.